src/jqLite.js

```diff
@@ -309,13 +303,10 @@ function jqLiteClone(element) {
 }
 
 function jqLiteDealoc(element, onlyDescendants) {
-  if (!onlyDescendants) jqLiteRemoveData(element);
+  if (!onlyDescendants && jqLiteAcceptsData(element)) jqLite.cleanData([element]);
```

Comment:
@gkalpak Let me elaborate why it shouldn't break anything now. The only situation where we would fire `jqLiteRemoveData(element)` before and not fire `jqLite.cleanData([element])` here now is when `jqLiteAcceptsData(element)` is `false`. But if you then look at `jqLiteRemoveData` source, it changes anything only if `element.ng339` is defined. This property is only set to something non-`undefined` inside `jqLiteExpandoStore` and only if the second passed argument is truthy. That, in turn, happens in two places only: one is inside `jqLiteData`, the other one is in `jqLite#on`. In both cases this setter is guarded by the `jqLiteAcceptsData(element)` being truthy which contradicts our assumptions.

Therefore, there should be no observable differences to external code here.